src/api/course/index.ts

```diff
@@ -274,13 +275,8 @@ export class CourseApi {
 
   /** Clear destination / route (exposed to plugins) */
   async clearDestination(persistState?: boolean): Promise<void> {
-    this.courseInfo.startTime = null
-    this.courseInfo.targetArrivalTime = null
-    this.courseInfo.activeRoute = null
-    this.courseInfo.nextPoint = null
-    this.courseInfo.previousPoint = null
+    this.courseInfo = NO_COURSE_INFO
     this.emitCourseInfo(!persistState)
-    this.cmdSource = null
```

Comment:
Clearing the destination should set  `this.cmdSource = null`.
Other than this I'm OK with the changes.